clang/utils/generate_unsupported_in_drivermode.py

```diff
@@ -0,0 +1,534 @@
+    if parent_field is None:
+        return set()
+
+    parent_name = parent_field["def"]
+    return {parent_name} | collect_transitive_groups(parent_name, options_dictionary)
+
+
+def get_visibility(option):
+    """Get a list of drivers that a given option is exposed to
+    option: The option object from Options.td
+    Return: Set that contains the visibilities of the given option
+    """
+    visibility_set = set()
+    # Check for the option's explicit visibility
+    for visibility in options_dictionary[option]["Visibility"]:
+        if visibility is not None:
+            visibility_set.add(visibility["def"])
+
+    # Check for the option's group's visibility
+    group_set = collect_transitive_groups(option, options_dictionary)
+    if group_set:
+        for group_name in group_set:
+            for visibility in options_dictionary[group_name]["Visibility"]:
+                visibility_set.add(visibility["def"])
+
+    return visibility_set
+
+
+def get_lit_test_note(test_visibility):
+    """Return the note to be included at the start of the Lit test file
+    test_visibility: Any VISIBILITY_* variable. VISIBILITY_DEFAULT will return the .c formatted test note.
+    All other will return the .f90 formatted test note
+    """
+    test_prefix = SLASH_SLASH if test_visibility == VISIBILITY_DEFAULT else EXCLAMATION
+
+    return (
+        f"{test_prefix}NOTE: This lit test was automatically generated to validate "
+        "unintentionally exposed arguments to various driver flavours.\n"
+        f"{test_prefix}NOTE: To make changes, see llvm-project/clang/utils/generate_unsupported_in_drivermode.py"
+        + " from which it was generated.\n"
+        f"{test_prefix}NOTE: Regenerate this Lit test with the following:\n"
+        f"{test_prefix}NOTE: python generate_unsupported_in_drivermode.py "
+        "--options-td-dir llvm-project/clang/include/clang/Driver --llvm-include-dir llvm-project/llvm/include --llvm-tblgen llvm-project/build/bin/llvm-tblgen\n\n"
+    )
+
+
+def write_lit_test(test_path, test_visibility):
+    """Write the Lit tests to file
+    test_path: File write path
+    test_visibility: VISIBILITY_DEFAULT, VISIBILITY_FLANG, or VISIBILITY_FC1 which indicates whether to write
+    to the main Lit test file, the flang test file, or the flang -fc1 test file
+    """
+    lit_file = open(test_path, "w")
+
+    lit_file.write(get_lit_test_note(test_visibility))
+    batch_size = 100
+
+    for visibility, driver_data in driver_data_dict.items():
+        is_flang_pair = visibility == VISIBILITY_FLANG or visibility == VISIBILITY_FC1
+
+        if (
+            (test_visibility == VISIBILITY_FLANG and visibility != VISIBILITY_FLANG)
+            or (test_visibility == VISIBILITY_FC1 and visibility != VISIBILITY_FC1)
+            or (test_visibility == VISIBILITY_DEFAULT and is_flang_pair)
+        ):
+            continue
+
+        comment_str = EXCLAMATION if is_flang_pair else SLASH_SLASH
```

Comment:
```suggestion
        comment_str = "! " if is_flang_pair else "// "
```
And remove `SLASH_SLASH` and `EXCLAMATION`